controllers/machine_controller.go

```diff
@@ -277,7 +277,7 @@ func (r *MachineReconciler) reconcileDelete(ctx context.Context, cluster *cluste
 	log := ctrl.LoggerFrom(ctx, "cluster", cluster.Name)
 
 	err := r.isDeleteNodeAllowed(ctx, cluster, m)
-	isDeleteNodeAllowed := err == nil //nolint:ifshort
+	isDeleteNodeAllowed := err == nil
```

Comment:
I was getting some conflicting linter errors on this file. nolintlint told me the tag wasn't being used (and make lint-fix removed it). ifshort seems to flag it sometimes though. Changed it here to see if it will pass, but I can remove it if wanted.

Reply:
this is a flake happening on mac, you can ignore it

Reply:
Interesting - I'll definitely exclude it from this PR, but it has the irritating impact that make lint-fix doesn't work as intended right now.

Reply:
100 % agreed, I got bitten in another PR but did not had time to investigate yet

Reply:
Same here. Will be fun to debug at some point :)

Reply:
I'll open an issue and give it a try. I already have the setup to run golangci-lint via the debugger

Reply:
https://github.com/kubernetes-sigs/cluster-api/issues/5652